File: samples/Linux/netcore_for_Linux_x64_only/netcore1.0/TableManagerSamples/CreateEntryOrder/src/ResponseListener.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading;
using fxcore2;

namespace CreateEntryOrder
{
    class ResponseListener : IO2GResponseListener
    {
        private string mRequestID;
        private EventWaitHandle mSyncResponseEvent;

        /// <summary>
        /// ctor
        /// </summary>
        public ResponseListener()
        {
            mRequestID = string.Empty;
            mSyncResponseEvent = new EventWaitHandle(false, EventResetMode.AutoReset);
        }

        public void SetRequestID(string sRequestID)
        {
            mRequestID = sRequestID;
        }

        public bool WaitEvents()
        {
            return mSyncResponseEvent.WaitOne(30000);
        }

        public void StopWaiting()
        {
            mSyncResponseEvent.Set();
        }

        #region IO2GResponseListener Members

        // Implementation of IO2GResponseListener interface public method onRequestCompleted
        public void onRequestCompleted(string sRequestID, O2GResponse response)
        {
        }

        // Implementation of IO2GResponseListener interface public method onRequestFailed
        public void onRequestFailed(string sRequestID, string sError)
        {
            if (mRequestID.Equals(sRequestID))
            {
                Console.WriteLine("Request failed: " + sError);
                StopWaiting();
            }
        }

        // Implementation of IO2GResponseListener interface public method onTablesUpdates
        public void onTablesUpdates(O2GResponse response)
        {
        }

        #endregion


    }
}

File: samples/Linux/java/TableManagerSamples/LockUpdates/src/ResponseListener.java
package lockupdates;

import java.util.concurrent.Semaphore;
import java.util.concurrent.TimeUnit;

import com.fxcore2.*;

public class ResponseListener implements IO2GResponseListener {
    private String mRequestID;
    private Semaphore mSemaphore;

    // ctor
    public ResponseListener() {
        mRequestID = "";
        mSemaphore = new Semaphore(0);
    }

    public void setRequestID(String sRequestID) {
        mRequestID = sRequestID;
    }

    public boolean waitEvents() throws Exception {
        return mSemaphore.tryAcquire(30, TimeUnit.SECONDS);
    }

    public void stopWaiting() {
        mSemaphore.release();
    }

    // Implementation of IO2GResponseListener interface public method onRequestCompleted
    public void onRequestCompleted(String sRequestID, O2GResponse response) {
    }

    // Implementation of IO2GResponseListener interface public method onRequestFailed
    public void onRequestFailed(String sRequestID, String sError) {
        if (mRequestID.equals(sRequestID)) {
            System.out.println("Request failed: " + sError);
            stopWaiting();
        }
    }

    // Implementation of IO2GResponseListener interface public method onTablesUpdates
    public void onTablesUpdates(O2GResponse response) {
    }
}
